spicy/runtime/include/util.h

```diff
@@ -31,4 +33,22 @@ extern const hilti::rt::Vector<
     std::optional<std::tuple<hilti::rt::integer::safe<uint64_t>, std::optional<hilti::rt::integer::safe<uint64_t>>>>>*
 get_offsets_for_unit(const hilti::rt::type_info::Struct& struct_, const hilti::rt::type_info::Value& value);
 
+/** Confirm a unit in try mode. */
+template<typename U>
+inline void confirm(U& p) {
+    p.__try_mode.reset();
```

Comment:
`__try_mode` is an `optional<ParseError>` which is only set if the unit is actually in trial mode; otherwise it is `none` and `reset` should be a no-op.

I'll leave this as is.

Reply:
Actually I was more concerned about the hook execution that follows right after this line.

Reply:
Good point, will make this conditional.